### PR TITLE
feat(Option): Support OkOrElse with materialized error

### PR DESCRIPTION
**Pull request by skarllot**
```csharp
some.OkOr("Err")
    .Should()
    .BeOk(123);
```

### CODE REVIEW

File: src/FxKit/Option/Option.NaturalTransformations.cs [hunk 1]
```diff
@@ -24,9 +25,30 @@ public static partial class Option
...
+    /// </summary>
+    /// <typeparam name="T">The type of the value in the result.</typeparam>
+    /// <typeparam name="R">The type of the error in the result.</typeparam>
+    /// <param name="source">The input <see cref="Option{T}"/>.</param>
+    /// <param name="invalidValue">The default error value to return if the option is empty.</param>
+    /// <returns>A <see cref="Result{TOk,TErr}"/> containing either the value or the specified error.</returns>
+    [DebuggerHidden]
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    [GenerateTransformer]
+    public static Result<T, R> OkOrElse<T, R>(
```
Comment (jeffijoe) [inline src/FxKit/Option/Option.NaturalTransformations.cs:44]: This should be named `OkOr` to be consistent with the other names that have both a materialized and a selector fallback variant.

Comment (skarllot) [reply src/FxKit/Option/Option.NaturalTransformations.cs]: Nice, fixed

File: src/FxKit/Option/Option.NaturalTransformations.cs [hunk 2]
```diff
@@ -9,13 +9,14 @@ namespace FxKit;
...
     /// </summary>
-    /// <param name="source"></param>
-    /// <param name="selector"></param>
-    /// <typeparam name="T"></typeparam>
-    /// <typeparam name="R"></typeparam>
-    /// <returns></returns>
+    /// <typeparam name="T">The type of the value in the result.</typeparam>
+    /// <typeparam name="R">The type of the error in the result.</typeparam>
+    /// <param name="source">The input <see cref="Option{T}"/>.</param>
+    /// <param name="selector">A function that provides the default error value to return if the option is empty.</param>
```
Comment (jeffijoe) [inline src/FxKit/Option/Option.NaturalTransformations.cs:18]: I don't think the `default` qualifier is needed here.

```suggestion
    /// <param name="selector">A function that provides the error value to return if the option is empty.</param>
```

File: src/FxKit/Option/Option.NaturalTransformations.cs [hunk 3]
```diff
@@ -24,9 +25,30 @@ public static partial class Option
...
+            : Result<T, R>.Err(selector());
+
+    /// <summary>
+    /// Returns a result containing the value from an <see cref="Option{T}"/> if it exists.
+    /// Otherwise, returns a result with the specified error value.
+    /// </summary>
+    /// <typeparam name="T">The type of the value in the result.</typeparam>
+    /// <typeparam name="R">The type of the error in the result.</typeparam>
+    /// <param name="source">The input <see cref="Option{T}"/>.</param>
+    /// <param name="invalidValue">The default error value to return if the option is empty.</param>
```
Comment (jeffijoe) [inline src/FxKit/Option/Option.NaturalTransformations.cs:39]: Same here:

```suggestion
    /// <param name="invalidValue">The error value to return if the option is empty.</param>
```

File: src/FxKit.Tests/UnitTests/Option/Option.NaturalTransformationTests.cs [hunk 4]
```diff
@@ -25,6 +25,21 @@ public void OkOrElse_ReturnsTheExpectedValue()
             .BeErr("Err");
     }
 
+    [Test]
+    public void OkOr_ReturnsTheExpectedValue()
+    {
+        Option<int> some = Some(123);
```
Comment (jeffijoe) [inline src/FxKit.Tests/UnitTests/Option/Option.NaturalTransformationTests.cs:31]: Here's an example of where implicit casts should work 😄 

```suggestion
        Option<int> some = 123;
```

File: src/FxKit/Option/Option.NaturalTransformations.cs (hunk 2 above)
Comment (skarllot) [reply src/FxKit/Option/Option.NaturalTransformations.cs]: Done

File: src/FxKit/Option/Option.NaturalTransformations.cs (hunk 3 above)
Comment (skarllot) [reply src/FxKit/Option/Option.NaturalTransformations.cs]: Done

File: src/FxKit.Tests/UnitTests/Option/Option.NaturalTransformationTests.cs (hunk 4 above)
Comment (skarllot) [reply src/FxKit.Tests/UnitTests/Option/Option.NaturalTransformationTests.cs]: Sometimes it works 😅

**Review by jeffijoe — APPROVED** [review]
Thank you!

**Pull request merged**